src/test/java/teammates/sqlui/webapi/JsonResultTest.java

```diff
@@ -0,0 +1,93 @@
+package teammates.sqlui.webapi;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.servlet.http.Cookie;
+
+import org.testng.annotations.Test;
+
+import teammates.test.BaseTestCase;
+import teammates.test.MockHttpServletResponse;
+import teammates.ui.output.MessageOutput;
+import teammates.ui.webapi.JsonResult;
+
+/**
+ * SUT: {@link JsonResult}.
+ */
+public class JsonResultTest extends BaseTestCase {
+    private static final String MESSAGE = "output message";
+    private static final String EMPTY_MESSAGE = "";
+    private final MessageOutput messageOutput = new MessageOutput(MESSAGE);
+    private final MessageOutput emptyMessageOutput = new MessageOutput(EMPTY_MESSAGE);
+    private final List<Cookie> cookies = new ArrayList<>();
+    private final List<Cookie> emptyCookies = new ArrayList<>();
+    private final Cookie testCookie = new Cookie("cookieName", "cookieValue");
+
+    @Test
+    public void testExecute_sendStringMessageReceivesMessage_shouldSucceed() throws Exception {
+        JsonResult result = new JsonResult(MESSAGE);
+        executeSuccessfulJsonResultTest(result);
+    }
+
+    @Test
+    public void testExecute_sendMessageOutputReceivesMessage_shouldSucceed() throws Exception {
+        JsonResult result = new JsonResult(messageOutput);
+        executeSuccessfulJsonResultTest(result);
+    }
+
+    private void executeSuccessfulJsonResultTest(JsonResult result) throws Exception {
```

Comment:
i like this abstraction! perhaps we can make it more general by adding parameters for `expectedNumberOfCookiesInResult` and `expectedMessageInResult` and `expectedNumberOfCookiesInResponse`, so that other test cases can use it too?

Reply:
maybe we can change the method name to `verifyJsonResult`? to be consistent with other private verification methods in other test files